crates/polars-plan/src/dsl/mod.rs

```diff
@@ -1263,7 +1296,17 @@ impl Expr {
                         fn_params: options.fn_params.clone(),
                     };
 
-                    rolling_fn(s, options).map(Some)
+                    match by_flag {
+                        IsSorted::Ascending => rolling_fn(&series, options).map(Some),
+                        IsSorted::Descending => {
+                            Ok(rolling_fn(&series, options)?.reverse()).map(Some)
+                        },
+                        IsSorted::Not => {
+                            let res = rolling_fn(&series, options)?;
+                            let indices = &original_indices.unwrap().arg_sort(Default::default());
```

Comment:
This should not be another sort and take, instead this should be a scatter: `out[sorting_indices[i]] = res[i]`. We currently lack a good scatter kernel though, as far as I'm aware. So this may be blocked until I get to that.

Reply:
thanks - OK marking as blocked then

Reply:
hi @orlp - just checking whether a scatter kernel might be on the horizon, and if not whether you have suggestions on what to do instead

Reply:
@MarcoGorelli We don't have a scatter kernel yet, but we do have `PolarsDataType::ZeroablePhysical`, which allows you to write a reasonably efficient scatter in a couple lines of code.

You can look in `gather_skip_nulls.rs` for an example, but basically you do

```rust
let mut out: Vec<T::ZeroablePhysical<'a>> = zeroed_vec(len);
unsafe {
    for i in 0..n {
        let out_idx = *sorted_indices.get_unchecked(i);
        *out.get_unchecked_mut(out_idx) = res.get(i).into();
    }
}
let out_arr = T::Array::from_zeroable_vec(out, dtype);
```

Reply:
Depending on the behavior of nulls you may also have to construct a null bitmap.

Reply:
Thanks

Noob question but where would I get `T` from in this function? Do I first need to refactor and make `finish_rolling` generic?